kiauh/components/klipper_firmware/menus/klipper_build_menu.py

```diff
@@ -10,34 +10,122 @@
 
 import textwrap
 from typing import List, Set, Type
+from os import path, listdir, mkdir
+from shutil import copyfile
 
 from components.klipper import KLIPPER_DIR
 from components.klipper_firmware.firmware_utils import (
     run_make,
     run_make_clean,
     run_make_menuconfig,
 )
-from core.logger import Logger
+from components.klipper_firmware.flash_options import FlashOptions
+from core.logger import DialogType, Logger
 from core.menus import Option
-from core.menus.base_menu import BaseMenu
+from core.menus.base_menu import BaseMenu, print_back_footer
 from core.types.color import Color
 from utils.sys_utils import (
     check_package_install,
     install_system_packages,
     update_system_package_lists,
 )
+from utils.input_utils import get_confirm, get_string_input, get_selection_input
 
 
 # noinspection PyUnusedLocal
 # noinspection PyMethodMayBeStatic
-class KlipperBuildFirmwareMenu(BaseMenu):
+class KlipperKConfigMenu(BaseMenu):
     def __init__(self, previous_menu: Type[BaseMenu] | None = None):
+        super().__init__()
+        self.title = "Firmware Config Menu"
+        self.title_color = Color.CYAN
+        self.previous_menu: Type[BaseMenu] | None = previous_menu
+        self.flash_options = FlashOptions()
+        self.kconfigs_dirname = path.expanduser("~/klipper/kconfigs")
+        self.kconfig_default = path.expanduser("~/klipper/.config")
+        self.kconfig = self.kconfig_default if not path.isdir(self.kconfigs_dirname) else None
+
+    def run(self) -> None:
```

Comment:
I feel like there is an issue with this run method not being called. I checked out your branch locally on my machine to try it out on my WSL instance, but when i go into the advanced menu and select 1, nothing happens and the advanced menu reloads. It seems like as if `KlipperKConfigMenu().run()` in `advanced_menu.py` would do nothing. I cannot explain why at the moment. I tried debugging it for a bit, but with no success. When i comment out that line so that the `KlipperBuildFirmwareMenu` is called, it works again and the Menu is presented, but as soon as the KlipperKConfigMenu should be run before that, it simply goes directly back into the AdvancedMenu.

Reply:
yeah my bad, fixed this